core/ExecutePipe.cpp

```diff
@@ -1,313 +1,145 @@
-        if(SPARTA_EXPECT_FALSE(ex_inst->isTransfer()))
-        {
-            if(ex_inst->getPipe() == InstArchInfo::TargetPipe::I2F)
-            {
-                // Integer source -> FP dest -- need to mark the appropriate destination SB
-                sparta_assert(reg_file_ == core_types::RegFile::RF_INTEGER,
-                              "Got an I2F instruction in an ExecutionPipe that does not source the integer RF: " << ex_inst);
-                const auto & dest_bits = ex_inst->getDestRegisterBitMask(core_types::RegFile::RF_FLOAT);
-                scoreboard_views_[core_types::RegFile::RF_FLOAT]->setReady(dest_bits);
-            }
-            else {
-                // FP source -> Integer dest -- need to mark the appropriate destination SB
-                sparta_assert(ex_inst->getPipe() == InstArchInfo::TargetPipe::F2I,
-                              "Instruction is marked transfer type, but I2F nor F2I: " << ex_inst);
-                sparta_assert(reg_file_ == core_types::RegFile::RF_FLOAT,
-                              "Got an F2I instruction in an ExecutionPipe that does not source the Float RF: " << ex_inst);
-                const auto & dest_bits = ex_inst->getDestRegisterBitMask(core_types::RegFile::RF_INTEGER);
-                scoreboard_views_[core_types::RegFile::RF_INTEGER]->setReady(dest_bits);
-            }
-        }
-        else {
-            const auto & dest_bits = ex_inst->getDestRegisterBitMask(reg_file_);
-            scoreboard_views_[reg_file_]->setReady(dest_bits);
-        }
-
-        // Testing mode to inject random branch misprediction to stress flushing mechanism
-        if (enable_random_misprediction_)
-        {
-            if (ex_inst->isBranch() && (std::rand() % 20) == 0)
-            {
-                ILOG("Randomly injecting a mispredicted branch: " << ex_inst);
-                FlushManager::FlushingCriteria criteria(FlushManager::FlushCause::MISPREDICTION, ex_inst);
-                out_execute_flush_.send(criteria);
-            }
-        }
-
-        // We're not busy anymore
-        unit_busy_ = false;
-
-        // Count the instruction as completely executed
-        ++total_insts_executed_;
-
-        // Schedule issue if we have instructions to issue
-        if (ready_queue_.size() > 0) {
-            issue_inst_.schedule(sparta::Clock::Cycle(0));
-        }
-
-        // Schedule completion
-        complete_inst_.preparePayload(ex_inst)->schedule(1);
+namespace olympia {
+const char ExecutePipe::name[] = "exe_pipe";
+
+ExecutePipe::ExecutePipe(sparta::TreeNode *node,
+                         const ExecutePipeParameterSet *p)
+    : sparta::Unit(node),
+      ignore_inst_execute_time_(p->ignore_inst_execute_time),
+      execute_time_(p->execute_time),
+      enable_random_misprediction_(p->enable_random_misprediction),
+      reg_file_(olympia::coreutils::determineRegisterFile(node->getGroup())),
+      collected_inst_(node, node->getName()) {
```

Comment:
Do you mind putting the `{` on the next line for readability?

Reply:
Resolved in [6b0e363](https://github.com/riscv-software-src/riscv-perf-model/pull/140/commits/6b0e3633c1b7331170afda8920f596f9fd16b233)